build_tools/build_libtorch.sh

```diff
@@ -102,19 +104,39 @@ fi
 
 checkout_pytorch() {
   if [[ ! -d "$PYTORCH_ROOT" ]]; then
-    git clone https://github.com/pytorch/pytorch "$PYTORCH_ROOT"
+    git clone --depth 1 --single-branch --branch "${PYTORCH_BRANCH}" https://github.com/pytorch/pytorch "$PYTORCH_ROOT"
   fi
   cd "$PYTORCH_ROOT"
-  git fetch --all
-  git checkout origin/"${PYTORCH_BRANCH}"
-  git submodule update --init --recursive
+  for dep in protobuf pocketfft cpuinfo FP16 psimd fmt sleef pybind11 onnx flatbuffers foxi; do
+    git submodule update --init --depth 1 -- third_party/$dep
+  done
 }
 
 build_pytorch() {
   cd "$PYTORCH_ROOT"
   # Uncomment the next line if you want to iterate on source builds
-  python setup.py clean
-
+  # python setup.py clean
+
+  sed -i.bak -E 's/^[[:space:]]+check_submodules()/#check_submodules()/g' setup.py
+  # remove after https://github.com/pytorch/pytorch/issues/81178 is fixed
+  sed -i.bak -E 's/namespace ao {/namespace ao {\n#include <ATen\/native\/ao_sparse\/quantized\/cpu\/packed_params.h>/g' aten/src/ATen/native/ao_sparse/quantized/cpu/qlinear_deserialize.cpp
+  sed -i.bak -E 's/namespace ao {/namespace ao {\n#include <ATen\/native\/ao_sparse\/quantized\/cpu\/packed_params.h>/g' aten/src/ATen/native/ao_sparse/quantized/cpu/qlinear_serialize.cpp
```

Comment:
dirty dirty hack but necessary until https://github.com/pytorch/pytorch/issues/81178 is resolved

Reply:
I hit this problem and the fix works.

Reply:
you will need to escape the `{` with `\{`

```
sed -i.bak -E 's/namespace ao \{/namespace ao \{\n#include <ATen\/native\/ao_sparse\/quantized\/cpu\/packed_params.h>/g' aten/src/ATen/native/ao_sparse/quantized/cpu/qlinear_deserialize.cpp
```

Reply:
https://github.com/llvm/torch-mlir/runs/7277257272?check_suite_focus=true#step:4:676

Reply:
yea portability between zsh (on the m1) and bash (on build machines) was what I wasn't sure about.